Tweak: AI - Show upgrade badge on context menu if no subscription [ED-12643]

matipojo opened the pull request.

matipojo committed:
New: AI Container Variations [ED-12554]

matipojo committed:
Cleanup

matipojo committed:
Fixes

matipojo committed:
Cleanup & fixes

matipojo committed:
Merge remote-tracking branch 'elementor/main' into ED-12554-ai-container-variations

matipojo committed:
Cleanup

matipojo committed:
Fix thumbnail ratio

matipojo committed:
Lint

matipojo committed:
Review Fixes

matipojo committed:
Review Fixes

matipojo committed:
Review Fixes

matipojo committed:
Merge remote-tracking branch 'elementor/main' into ED-12554-ai-container-variations

matipojo committed:
Review Fixes

matipojo committed:
Merge remote-tracking branch 'elementor/main' into ED-12554-ai-container-variations

matipojo committed:
Lint

matipojo committed:
Fix: Server can't read some arrays

Send full body as JSON fixed it

matipojo committed:
Cleanup

matipojo committed:
Tweak: AI - Show upgrade badge on context menu if no subscription

matipojo committed:
Merge remote-tracking branch 'elementor/main' into ED-12554-ai-container-variations-upgrade-badge

matipojo committed:
Extract style from js

matipojo committed:
Cleanup and docs

StyleShit commented on modules/ai/assets/js/editor/layout-module.js:
```diff
@@ -58,7 +57,27 @@ export default class Module extends elementorModules.editor.utils.Module {
+
+		if ( ! ElementorAiConfig?.usage?.hasAiSubscription ) {
+			const svg = `
+			<svg viewBox="0 0 24 24">
+				<path d="M12 5.25C12.2508 5.25 12.485 5.37533 12.6241 5.58397L16.1703 10.9033L20.5315 7.41435C20.7777
+				7.21743 21.1207 7.19544 21.39 7.35933C21.6592 7.52321 21.7973 7.83798 21.7355 8.14709L19.7355
+				18.1471C19.6654 18.4977 19.3576 18.75 19 18.75H5.00004C4.64253 18.75 4.33472 18.4977 4.26461
+				18.1471L2.2646 8.14709C2.20278 7.83798 2.34084 7.52321 2.61012 7.35933C2.8794 7.19544 3.22241
+				7.21743 3.46856 7.41435L7.82977 10.9033L11.376 5.58397C11.5151 5.37533 11.7493 5.25 12 5.25ZM12
+				7.35208L8.62408 12.416C8.50748 12.5909 8.32282 12.7089 8.1151 12.7411C7.90738 12.7734 7.69566
+				12.717 7.53152 12.5857L4.13926 9.87185L5.61489 17.25H18.3852L19.8608 9.87185L16.4686 12.5857C16.3044
+				12.717 16.0927 12.7734 15.885 12.7411C15.6773 12.7089 15.4926 12.5909 15.376 12.416L12 7.35208Z">
+				</path>
+    		</svg>`;
```
hmm
is there a way to reuse the SVG from `@elementor/icons`?

matipojo replied:
It's outside of React.

StyleShit commented on modules/ai/connect/ai.php:
```diff
@@ -38,6 +38,17 @@ public function get_usage() {
+	public function get_cached_usage() {
+		$cache_key = 'elementor_ai_usage';
+		$cache_time = 24 * HOUR_IN_SECONDS;
+		$usage = get_site_transient( $cache_key );
+		if ( ! $usage ) {
+			$usage = $this->get_usage();
+			set_site_transient( $cache_key, $usage, $cache_time );
+		}
+		return $usage;
```
Let it breath 😅 

```suggestion
		$usage = get_site_transient( $cache_key );
		
		if ( ! $usage ) {
			$usage = $this->get_usage();
			set_site_transient( $cache_key, $usage, $cache_time );
		}
		
		return $usage;
```

StyleShit commented on modules/ai/module.php:
```diff
@@ -148,6 +148,8 @@ private function enqueue_main_script() {
 			[
 				'is_get_started' => User::get_introduction_meta( 'ai_get_started' ),
 				'connect_url' => $this->get_ai_connect_url(),
+				// Use a cached version, don't call the API on every editor load.
+				'usage' => $this->get_ai_app()->get_cached_usage(),
```
🔥

matipojo committed:
Update modules/ai/connect/ai.php

Co-authored-by: StyleShit <32631382+StyleShit@users.noreply.github.com>

matipojo committed:
Merge branch 'main' into ED-12554-ai-container-variations-upgrade-badge

matipojo committed:
Merge branch 'main' into ED-12554-ai-container-variations-upgrade-badge

Omerisra6 committed:
Internal: Add twentytwentyone theme to WP 6.4+ installs (#24227)

arielk committed:
Tweak: “Global Style Guide” experiment was promoted to a stable feature CP [ED-11808] (#24217)

### Includes: 
- Tweak: “Global Style Guide” experiment was promoted to a stable
feature
- Internal: Activate Optimize Image Loading experiment by default for
all sites
- Internal: Activate Optimized Gutenberg Loading experiment by default
for all sites
- Internal: Activate Inline Font Icons experiment by default for new
sites

---------

Co-authored-by: Dmytro Maziar <dmytrom@elementor.com>

KingYes committed:
New: Introducing Element Manager - Disable/enable widgets [ED-12546] (#24231)

Co-authored-by: Ariel Klikstein <ariel@elementor.com>

blackcolt committed:
Merge branch 'main' into ED-12554-ai-container-variations-upgrade-badge

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/elementor/elementor?pullRequest=24194) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/elementor/elementor?pullRequest=24194) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: matipojo<br/>:x: blackcolt<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/elementor/elementor?pullRequest=24194) it.</sub>

blackcolt committed:
Merge branch 'main' into ED-12554-ai-container-variations-upgrade-badge

Omerisra6 committed:
Tweak: Get super global changed to handle multiple files [ED-2192]  (#24248)

matipojo committed:
Merge branch 'main' into ED-12554-ai-container-variations-upgrade-badge

matipojo committed:
Merge remote-tracking branch 'elementor/3.18' into ED-12554-ai-container-variations-upgrade-badge

matipojo committed:
Lint

matipojo committed:
Merge branch 'main' into ED-12554-ai-container-variations-upgrade-badge
